mlos_bench/mlos_bench/tunables/tunable.py

```diff
@@ -109,11 +109,13 @@ def value(self, value):
         """
         Set the current value of the tunable.
         """
-        if not self.is_valid(value):
+        # We need this coercion for the values produced by some optimizers
+        # (e.g., scikit-optimize) and for data restored from certain storage
+        # systems (where values can be strings).
+        coerced_value = self._TYPE[self._type](value)
```

Comment:
I guess this may raise a `ValueError` if the coersion can't be done?
Should we handle that and try and provide a more useful error message when that happens?
Maybe this also addresses my question about checking for `int("1.0")` being allowed but `int("1.1")` not being allowed.

Reply:
Yes, that throws `ValueError` when we can't convert, including the case of `int("1.1")`. I'll add a user-friendly detailed log message if that happens, but I think we have to rethrow and crash anyway - bad value here means that either our scripts are wrong, or we had saved some invalid data in the storage